Location.py

```diff
@@ -67,13 +67,17 @@ def set_rule(self, lambda_rule):
         self.access_rules = [lambda_rule]
 
 
-    def can_fill(self, state, item, check_access=True):
+    def can_fill(self, state, item, check_access=True, extra_location_checks=[]):
```

Comment:
```suggestion
    def can_fill(self, state, item, check_access=True, extra_location_checks=()):
```
Avoid mutable defaults.